client/lib/csv/index.js

```diff
@@ -0,0 +1,65 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import moment from 'moment';
+import { saveAs } from 'browser-filesaver';
+
+function getCSVHeaders( headers ) {
+	return headers.map( header => header.label ).join( ',' );
+}
+
+function getCSVRows( rows ) {
+	return rows.map( row => row.map( rowItem => rowItem.value ).join( ',' ) ).join( '\n' );
+}
+
+function getCSVTotals( headers, totals ) {
+	return headers.map( header => get( totals, header.key, '-' ) ).join( ',' );
+}
+
+/**
+ * Generates a CSV string from table contents
+ *
+ * @param   {Array.<Object>}        [headers=[]]    Object with table header information
+ * @param   {Array.Array.<Object>}  [rows=[]]       Object with table rows information
+ * @param   {Object}                [totals=[]]     Object with totals information
+ * @returns {String}                                Table contents in a CSV format
+ */
+export function generateCSVDataFromTable( headers = [], rows = [], totals = [] ) {
+	return [ getCSVHeaders( headers ), getCSVRows( rows ), getCSVTotals( headers, totals ) ].join(
+		'\n'
+	);
+}
+
+/**
+ * Generates a file name for CSV files based on the provided name, the current date
+ * and the provided params, which are all appended with hyphens.
+ *
+ * @param   {String}  [name='']     Name of the file
+ * @param   {Object}  [params={}]   Object of key-values to append to the file name
+ * @returns {String}                Formatted file name
+ */
+export function generateCSVFileName( name = '', params = {} ) {
+	const fileNameSections = [
+		name,
+		moment().format( 'YYYY-MM-DD' ),
```

Comment:
really like the addition of a timestamp to the filename, for cluttered desktop users like me, this is a very nice subtle feature 👍